app/Team.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Team extends Model
+{
+    protected $fillable = [
+        'name',
+        'amount_players',
+        'logo_url',
+        'user_id'
+    ];
+
+    public function players()
+    {
+        return $this->belongsToMany('App\Player')
+    }
```

Comment:
Creo que falta la relación con tournamente.  $this->belongsToMany('App\Tournament');